webapp/src/main/java/org/apache/atlas/notification/ImportTaskListenerImpl.java

```diff
@@ -153,7 +162,48 @@ public void onCompleteImportRequest(String importId) {
         }
     }
 
-    private void startNextImportInQueue() {
+    @PreDestroy
+    public void stopImport() {
+        LOG.info("Shutting down import processor...");
+
+        executorService.shutdown(); // Initiate an orderly shutdown
+
+        try {
+            if (!executorService.awaitTermination(30, TimeUnit.SECONDS)) {
+                LOG.warn("Executor service did not terminate gracefully within the timeout. Waiting longer...");
+
+                // Retry shutdown before forcing it
+                if (!executorService.awaitTermination(10, TimeUnit.SECONDS)) {
+                    LOG.warn("Forcing shutdown...");
+
+                    executorService.shutdownNow();
+                }
+            }
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+
+            LOG.error("Shutdown interrupted. Forcing shutdown...");
+
+            executorService.shutdownNow();
+        }
+
+        LOG.info("Import processor stopped.");
+    }
+
+    @VisibleForTesting
+    void startInternal() {
+        populateRequestQueue();
+
+        CompletableFuture.runAsync(this::startNextImportInQueue)
```

Comment:
When  `requestQueue` is empty, it is not necessary to call `startNextImportInQueue()`. This will save unnecessary attempts to load the next request - in `getNextImportFromQueue()`.